From f9caf95feab46d93a96763e7fb051003210617ff Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Wed, 3 Jun 2015 16:28:14 +0800
Subject: [PATCH 414/462] atomisp2: Fix resolution selection for SDV mode

The current resolution negotiation logic does not
ask for a sensor setting with DVS envelop in SDV
mode. This forces DVS to be disabled since no envelop
is available. This patch fixes this logic and asd
for a resolution with DVS envelop in SDV mode.

Change-Id: Iaf5e4305dc3d183fea4e89630a8ddf514bedab2b
Fixes-Issue: #JIMINAN-38335
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/58702
Validation-Android: Wei Tao <wei.tao@intel.com>
Approver: Bin B Han <bin.b.han@intel.com>
Reviewed-by: Bin B Han <bin.b.han@intel.com>
Build: IMIN (SC)
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c     | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
index 4572e6e..ce82270 100755
--- a/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
+++ b/drivers/external_drivers/camera/drivers/media/pci/atomisp2/atomisp_driver/atomisp_cmd.c
@@ -5084,8 +5084,7 @@ static void atomisp_get_dis_envelop(struct atomisp_sub_device *asd,
 		asd->params.video_dis_en = 0;
 
 	if (asd->params.video_dis_en &&
-	    asd->run_mode->val == ATOMISP_RUN_MODE_VIDEO &&
-	    !asd->continuous_mode->val) {
+	    asd->run_mode->val == ATOMISP_RUN_MODE_VIDEO) {
 		/* envelope is 20% of the output resolution */
 		/*
 		 * dvs envelope cannot be round up.
-- 
1.9.1

